CMakeLists.txt

```
@@ -0,0 +1,186 @@
+    MALLOC_IS_ALIGNED
+    NO_THREAD_CHECKS
+    NO_TIMING
+)
+
+target_include_directories(rubberband
+  PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/rubberband>
+      $<INSTALL_INTERFACE:rubberband>
+  PRIVATE
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+)
+
+target_link_libraries(rubberband
+  PRIVATE
+    Threads::Threads
+)
+
+# Handle FFT library
+if(FFTW3_INCLUDEDIR AND FFTW3_LIBRARY)
+  target_include_directories(rubberband
+    PRIVATE
+      ${FFTW3_INCLUDEDIR}
+  )
+  target_compile_definitions(rubberband
+    PRIVATE
+      HAVE_FFTW3
+      FFTW_DOUBLE_ONLY
+  )
+  target_link_libraries(rubberband
+    PUBLIC
+      ${FFTW3_LIBRARY}
+  )
+else()
+  target_compile_definitions(rubberband
+    PRIVATE
+      USE_KISSFFT
+  )
+  target_sources(rubberband
+    PRIVATE
+      src/kissfft/kiss_fft.c
+      src/kissfft/kiss_fftr.c
+  )
+endif()
+
+# Apple framework
+if(APPLE)
+  target_link_libraries(rubberband
+    PRIVATE
+      "${ACCELERATE_FRAMEWORK}"
+  )
+endif()
+
+# Handle resampling library
+if(TARGET samplerate)
+  target_compile_definitions(rubberband
+    PRIVATE
+      HAVE_LIBSAMPLERATE
+  )
+
+  target_link_libraries(rubberband
+    PRIVATE
+      samplerate
+  )
+elseif(SAMPLERATE_INCLUDEDIR AND SAMPLERATE_LIBRARY)
+  target_include_directories(rubberband
+    PRIVATE
+      ${SAMPLERATE_INCLUDEDIR}
+  )
+  target_compile_definitions(rubberband
+    PRIVATE
+      HAVE_LIBSAMPLERATE
+  )
+  target_link_libraries(rubberband
+    PUBLIC
+      ${SAMPLERATE_LIBRARY}
+  )
+else()
+  target_compile_definitions(rubberband
+    PRIVATE
+      USE_SPEEX
+  )
+  target_sources(rubberband
+    PRIVATE
+      src/speex/resample.c
+  )
+endif()
+
+### Source files ###
+target_sources(rubberband
+  PRIVATE
+    src/RubberBandStretcher.cpp
+    src/StretchCalculator.cpp
+    src/StretcherChannelData.cpp
+    src/StretcherImpl.cpp
+    src/StretcherProcess.cpp
+    src/audiocurves/CompoundAudioCurve.cpp
+    src/audiocurves/ConstantAudioCurve.cpp
+    src/audiocurves/HighFrequencyAudioCurve.cpp
+    src/audiocurves/PercussiveAudioCurve.cpp
+    src/audiocurves/SilentAudioCurve.cpp
+    src/audiocurves/SpectralDifferenceAudioCurve.cpp
+    src/base/Profiler.cpp
+    src/dsp/AudioCurveCalculator.cpp
+    src/dsp/FFT.cpp
+    src/dsp/Resampler.cpp
+    src/rubberband-c.cpp
+    src/system/Allocators.cpp
+    src/system/Thread.cpp
+    src/system/VectorOpsComplex.cpp
+    src/system/sysutils.cpp
+)
+
+### Example ###
+if(RUBBERBAND_EXAMPLE)
+  add_executable(rubberband_test main/main.cpp)
+  target_include_directories(rubberband_test PRIVATE src Threads::Threads)
+  target_link_libraries(rubberband_test PRIVATE rubberband -lsndfile)
+endif()
+
+### Installation ###
+
+install(
+  TARGETS rubberband
+  EXPORT rubberband-exports
+  RUNTIME DESTINATION lib
+  ARCHIVE DESTINATION lib)
```

On Windows, DLL files belong into `bin/`. This should cover all cases:

```suggestion
include(GNUInstallDirs)
install(
  TARGETS rubberband
  EXPORT rubberband-exports
  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
```